### PR TITLE
CSD-71 Migration configuration and fields for courses from explorecourses

### PR DESCRIPTION
**Pull request by pookmish**
# READY FOR REVIEW

# Summary
- Migration entities for course importers into opportunities.

# Need Review By (Date)
- 5/19

# Urgency
- Medium

# Steps to Test
1. `composer require su-hkku/cardinal_service_profile:dev-CSD-71`
1. `blt drupal:sync --site=cardinalservice -n`
1. run one of the migrations like `drush @cardinalservice.local mim cardcourses_educ`
1. in the admin, verify some courses were imported
1. view the course to verify alot of fields were populated.

# See Also
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)

### CODE REVIEW

**Comment by joegl** [comment]
Running the migration command c mpleted successfully with:
`[notice] Processed 53 items (45 created, 8 updated, 0 failed, 0 ignored) - done with 'cardcourses_educ'`

I checked 10 nodes -- lots of fields were populated. The following fields were left unpopulated on all nodes (not if working as designed or missing data):
      Date: Submissions Open, Application Deadline, Duration Start/End, 
      Text: Eligibility, Contact Name/Email/Phone
      Taxonomy Ref: Pathway, Placement Type, Sponsor
      Image: Image

File: cardinal_service_profile.inc [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+      continue;
+    }
+
+    list(, $from_month, $from_day) = explode('-', $deadline_range[0]['value']);
+    list(, $to_month, $to_day) = explode('-', $deadline_range[0]['end_value']);
+
+    // This term is configured for the range surrounding the deadline date field
+    // so we use that term id as the deadline term.
+    if ($month >= $from_month && $month <= $to_month && $day >= $from_day && $day <= $to_day) {
+      $node->set('su_opp_deadline_time', $tid);
```
Comment (joegl) [inline cardinal_service_profile.inc:43]: Should` su_opp_deadline_time` here be using the `su_opp_application_deadline` namespace?

Comment (pookmish) [reply cardinal_service_profile.inc:43]: the namespace is the `su_opp` part. `su` = stanford university, `opp` = opportunity

Comment (joegl) [reply cardinal_service_profile.inc:43]: I see below in the config the su_opp_deadline_time field was added. I will double check

Comment (joegl) [reply cardinal_service_profile.inc:43]: Yea namespace was poor word choice. I wasn't sure if it was the correct field to take in the value.

Comment (joegl) [reply cardinal_service_profile.inc:43]: And now I see that's done by design. Ignore this comment thread please.

**Comment by pookmish** [comment]
Yeah not all fields get populated. there are different importers that populate what they can.

**Pull request merged**